table/block_based_table_reader.h

```diff
@@ -717,12 +733,17 @@ class BlockBasedTableIterator : public InternalIteratorBase<TValue> {
   bool for_compaction_;
   BlockHandle prev_index_value_;
 
-  static const size_t kInitReadaheadSize = 8 * 1024;
+  // All the below fields control iterator readahead
+  static const size_t kInitAutoReadaheadSize = 8 * 1024;
   // Found that 256 KB readahead size provides the best performance, based on
-  // experiments.
-  static const size_t kMaxReadaheadSize;
-  size_t readahead_size_ = kInitReadaheadSize;
+  // experiments, for auto readahead. Experiment data is in PR #3282.
+  static const size_t kMaxAutoReadaheadSize = 256 * 1024;
+  // If read_options.readahead_size is not set,
+  static const int kMinNumFileReadsToStartAutoReadahead = 2;
+  size_t readahead_size_ = 0;
+  size_t max_readahead_size_ = 0;
```

Comment:
What's the point of setting a default that isn't make sense?

Reply:
It doesn't matter imo. 0 is still saner than leaving it uninitialized, as the initializations were done inside the constructor for these fields. But fixed now anyway.